models/repo/archiver.go

```diff
@@ -146,6 +146,13 @@ func FindRepoArchives(opts FindRepoArchiversOption) ([]*RepoArchiver, error) {
 // SetArchiveRepoState sets if a repo is archived
 func SetArchiveRepoState(repo *Repository, isArchived bool) (err error) {
 	repo.IsArchived = isArchived
-	_, err = db.GetEngine(db.DefaultContext).Where("id = ?", repo.ID).Cols("is_archived").NoAutoTime().Update(repo)
+
+	if isArchived {
+		repo.ArchivedUnix = timeutil.TimeStampNow()
+	} else {
+		repo.ArchivedUnix = timeutil.TimeStamp(0)
+	}
```

Comment:
Hmm, I'm inclined to discard this `else` block.
```suggestion
	}
```

Reply:
That would mean, that this is not updated when unarchiving a repo. So you end up with Repos that have a archive date but who are not archived.

Reply:
As I see it:
We don't care about the archived date as long as the repo isn't archived.
If an unarchived repo has an archived date, it should simply be ignored.

Reply:
The date can be archived through the API, so this will be visible to end users

Reply:
Okay, yes, I can understand that concern.
It certainly looks weird if the API returns both `archived_unix != 0` and `is_archived = false`.
But still, the current approach is not a good idea either.

Reply:
The API currently return `null`, if there is no archive date

Reply:
I know, as I said, this comment above is not meant for this suggestion anymore, this one is now resolved.